bin/runFixedPop

```diff
@@ -256,42 +260,50 @@ if __name__ == '__main__':
         dat_store.append('idx', pd.DataFrame([idx]))
 
         # Filter the bcm and bpp arrays according to user specified filters
-        bcm = filter_bpp_bcm(bcm, bpp, filters)
-        
-        # Filter out systems that don't match final kstars supplied
-        bcm_save_filtered = bcm.loc[(bcm.kstar_1.isin(kstar1_range)) & (bcm.kstar_2.isin(kstar2_range))]
-        
+        if len(bcm_filter) > 1:
+            bcm_filter = bcm_filter.append(filter_bpp_bcm(bcm, bpp, filters, kstar1_range, kstar2_range))
+            initC_filter = initC_filter.append(initCond.loc[initCond.bin_num.isin(bcm_filter.bin_num)])
+            bpp_filter = bpp_filter.append(bpp.loc[bpp.bin_num.isin(bcm_filter.bin_num)])
+        else:
+            bcm_filter = filter_bpp_bcm(bcm, bpp, filters, kstar1_range, kstar2_range)
+            initC_filter = initCond.loc[initCond.bin_num.isin(bcm_filter.bin_num)]
+            bpp_filter = bpp.loc[bpp.bin_num.isin(bcm_filter.bin_num)]
+
         # Run the match on short period binaries, such that we get a better sampling
         # for the highest signal systems
-        if (len(bcm_save_filtered) > 0):
+        n_bin_state = []
+        for bin_state in filters['binary_state']:
+            n_bin_state.append(len(bcm_filter.loc[bcm_filter.bin_state == bin_state]))
+        if all(i >= 50 for i in n_bin_state):
```

Comment:
I understand making sure that each binary state of interest has enough samples to be worth testing convergence on. However I wonder about the fact that this logic throws out iterations where we have have 49 systems that made it to bin state 0 1 and 2 each and the code says oh that's not enough and does not save them to the overall `bcm_save` variable, should we not still save them to the array but not calculate convergence? or am I missing something.

Reply:
@scottcoughlin2014 
I can't figure out how to quote the code, but starting on line 263 of [runFixedPop](https://github.com/COSMIC-PopSynth/COSMIC/pull/107/files#diff-30a06cad5f4b8c2bdc7fcbbad5e6305aR263) is where the `bcm_filter`, `initC_filter` and `bpp_filter` DataFrames are appended to until each `bin_state` has >50 values in it. Once this condition is fulfilled you then compute the convergence, `bcm_save` and `bpp_save` are appended to,  and `bcm_filter`, `initC_filter` and `bpp_filter` are emptied to be filled again.